jdaviz/app.py

```diff
@@ -721,52 +721,18 @@ def get_subsets_from_viewer(self, viewer_reference, data_label=None, subset_type
                                          cls=None)
         regions = {}
 
-        def _get_all_subregions(mask, spec_axis_data):
-            """
-            Return all subregions within a subset.
-
-            Parameters
-            ----------
-            mask : list
-                List of indices in spec_axis_data that are part of the subset.
-            spec_axis_data : list
-                List of spectral axis values.
-            Returns
-            -------
-            combined_spec_region : `~specutils.SpectralRegion`
-                SpectralRegion object containing all subregions of the subset.
-            """
-            if len(mask) == 0:
-                # Mask should only be 0 if ApplyROI is used to incorrectly
-                # create subsets via the API
-                raise ValueError("Mask has length 0, ApplyROI may have been used incorrectly")
-
-            current_edge = 0
+        def _get_all_subregions(spectral_axis_units, subset_label):
             combined_spec_region = None
-            for index in range(1, len(mask)):
-                # Find spot where mask == True is for a different region of the subset
-                # i.e. mask = [0, 1, 4, 5]
-                # mask[2] != mask[1] + 1
-                if mask[index] != mask[index - 1] + 1:
-                    subset_region = spec_axis_data[mask[current_edge]: mask[index - 1] + 1]
-                    if not combined_spec_region:
-                        combined_spec_region = SpectralRegion(min(subset_region),
-                                                              max(subset_region))
+            subset_plugin = self._jdaviz_helper.plugins['Subset Tools']
+            all_subregions = subset_plugin.get_all_subsets_with_subregions()
+            if subset_label in all_subregions:
+                for subregion in all_subregions[subset_label]:
+                    if combined_spec_region is None:
+                        combined_spec_region = SpectralRegion(subregion[0] * spectral_axis_units,
```

Comment:
So, I noticed that when you have compound subset with 2 sub-subsets, you get 3 subset state objects (I think 1 is the overall subset, and then 2 other for the actual components?). Do you really want the "overall" part because it includes the gap? 🤷

Reply:
I'm not sure I follow...here is a screenshot with a compound subset:
<img width="1396" alt="Screen Shot 2022-12-01 at 10 05 37 AM" src="https://user-images.githubusercontent.com/7179333/205087400-37ff1e22-6848-44a7-b95b-6a901e434e70.png">

Reply:
Hmmm... I'll have to double check on my side then.